object/virtual_device_list.go

```diff
@@ -455,7 +463,14 @@ func (l VirtualDeviceList) AssignController(device types.BaseVirtualDevice, c ty
 	d := device.GetVirtualDevice()
 	d.ControllerKey = c.GetVirtualController().Key
 	d.UnitNumber = new(int32)
-	*d.UnitNumber = l.newUnitNumber(c)
+
+	switch device.(type) {
```

Comment:
Thanks @BrianMMcClain , lgtm. One request, can you just add an offset param to `newUnitNumber`?  Then here would be something like:
```go
	offset := 0
	switch device.(type) {
	case types.BaseVirtualEthernetCard:
		offset = 7
	}
	*d.UnitNumber = l.newUnitNumber(c, offset)
```

Reply:
Done! Wasn't sure if modifying the existing method was better or not interfering with existing calls was better. Looks like that and the test was the only place that function is called though :)